src/v/kafka/client/produce_partition.h

```diff
@@ -34,43 +34,42 @@ class produce_partition {
     produce_partition(const configuration& config, consumer&& c)
       : _config{config}
       , _batcher{compression_from_str(config.produce_compression_type())}
-      , _timer{[this]() { try_consume(true); }}
+      , _timer{[this]() -> ss::future<> { co_await try_consume(true); }}
       , _consumer{std::move(c)} {}
 
-    ss::future<response> produce(model::record_batch&& batch) {
+    ss::future<ss::future<response>> produce(model::record_batch&& batch) {
```

Comment:
I guess this was as issue before but it seems weird that the current fiber would have to wait on `try_consume` before returning, when it mainly executes in the background in the context of a timer, with the exception of in one place `stop()`. I guess its fine for now but i'd vote on making try_consume just arm the timer and then the timer can have a small lambda that invokes `_consumer()` @BenPope

Reply:
The original implementation still waited for `try_consume` to complete, but it was synchronous.  I don't think the behavior, in relation to that, is changing.

Reply:
resolved offline, updates published in latest force-push